src/main/java/team9/ddang/family/service/FamilyServiceImpl.java

```diff
@@ -0,0 +1,284 @@
+                    return new InviteCodeResponse(family, existingInviteCode, ttl);
+                }
+            }
+        }
+
+        // 새로운 초대 코드 생성
+        String newInviteCode = generateInviteCode(family.getFamilyId());
+        redisTemplate.opsForValue().set("invite:" + newInviteCode, String.valueOf(family.getFamilyId()), Duration.ofMinutes(5));
+
+        return new InviteCodeResponse(family, newInviteCode, Duration.ofMinutes(5).toSeconds());
+    }
+
+
+    @Override
+    @Transactional
+    public FamilyResponse addMemberToFamily(String inviteCode, Member member) {
+        Member currentMember = findMemberByIdOrThrowException(member.getMemberId());
+
+        String familyIdStr = redisTemplate.opsForValue().get("invite:" + inviteCode);
+        if (familyIdStr == null) {
+            throw new IllegalArgumentException(FamilyExceptionMessage.INVALID_INVITE_CODE.getText());
+        }
+
+        if (currentMember.getFamily() != null) {
+            throw new IllegalArgumentException(FamilyExceptionMessage.MEMBER_ALREADY_IN_FAMILY.getText());
+        }
+
+        boolean hasDog = memberDogRepository.existsByMember(member);
+        if (hasDog) {
+            throw new IllegalArgumentException(FamilyExceptionMessage.MEMBER_DOG_FOUND.getText());
+        }
+
+        Long familyId = Long.valueOf(familyIdStr);
+
+        Family family = findFamilyByIdOrThrowException(familyId);
+
+        currentMember.updateFamily(family);
+
+        List<Dog> dogs = dogRepository.findAllByFamilyId(family.getFamilyId());
+
+        for (Dog dog : dogs) {
+            MemberDog memberDog = MemberDog.builder()
+                    .member(currentMember)
+                    .dog(dog)
+                    .build();
+            memberDogRepository.save(memberDog);
+        }
+
+        return new FamilyResponse(family);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public FamilyDetailResponse getMyFamily(Member member) {
+        Member currentMember = findMemberByIdOrThrowException(member.getMemberId());
+
+        if (currentMember.getFamily() == null) {
+            throw new IllegalArgumentException(FamilyExceptionMessage.MEMBER_NOT_IN_FAMILY.getText());
+        }
+
+        Family family = currentMember.getFamily();
+        List<GetDogResponse> dogs = dogRepository.findAllByFamilyId(family.getFamilyId())
+                .stream()
+                .map(dog -> new GetDogResponse(
+                        dog.getDogId(),
+                        dog.getName(),
+                        dog.getBreed(),
+                        dog.getBirthDate(),
+                        dog.getWeight(),
+                        dog.getGender(),
+                        dog.getProfileImg(),
+                        dog.getIsNeutered(),
+                        family.getFamilyId(),
+                        dog.getComment()
+                ))
+                .collect(Collectors.toList());
```

Comment:
GetDogResponse에 정적 팩토리 메소드 작성하여서 아래처럼 변경 할 수 있을 것 같습니다!

```suggestion
List<GetDogResponse> dogs = dogRepository.findAllByFamilyId(family.getFamilyId())
        .stream()
        .map(dog -> GetDogResponse.from(dog, family.getFamilyId()))
        .collect(Collectors.toList());
```
정적 팩토리 메소드
```public static GetDogResponse from(Dog dog, Long familyId) {
        return new GetDogResponse(
                dog.getDogId(),
                dog.getName(),
                dog.getBreed(),
                dog.getBirthDate(),
                dog.getWeight(),
                dog.getGender(),
                dog.getProfileImg(),
                dog.getIsNeutered(),
                familyId,
                dog.getComment()
        );
    }
```

Reply:
GetDogResponse를 작성한 게 제가 아니라서...😅
나중에 리펙토링 해볼게요!

Reply:
아하 그렇군요! 고생하셨습니다